tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/bigquery_tests_base.py

```diff
@@ -0,0 +1,243 @@
+                             True)
+
+        @classmethod
+        def tearDownClass(cls):
+            """
+            Remove the test dataset table(s).
+            """
+            for table in cls.fq_table_names + cls.fq_sandbox_table_names:
+                cls.client.delete_table(table)
+
+        def drop_rows(self, fq_table_name):
+            """
+            Helper function to drop table rows and assert the drop finished.
+
+            Raises an assertion error if the table records are not dropped.
+            Drops all rows.
+
+            :param fq_table_name: a fully qualified table name to drop all
+                records for
+            """
+            query = f"delete from {fq_table_name} where true"
+
+            response = self.client.query(query)
+
+            # start the job and wait for it to complete
+            self.assertIsNotNone(response.result())
+            self.assertIsNone(response.exception())
+
+        def tearDown(self):
+            """
+            Ensure the data is dropped from the table(s).
+            """
+            for table in self.fq_table_names + self.fq_sandbox_table_names:
+                self.drop_rows(table)
+
+        def load_test_data(self, sql_statements=None):
+            """
+            Add data to the tables for the rule to run on.
+
+            :param sql_statements: a list of sql statements to load for testing.
+            """
+
+            if not sql_statements or not isinstance(sql_statements, list):
+                raise RuntimeError("Provide parameter sql_statements.  The "
+                                   "parameter must be list of sql strings.")
+
+            for query in sql_statements:
+                response = self.client.query(query)
+                self.assertIsNotNone(response.result())
+                self.assertIsNone(response.exception())
+
+    class DropRowsTestBase(BigQueryTestBase):
+        """
+        Class that can be extended and used to test cleaning rules that drop row data.
+
+        This class defines a basic test that can be used by extending classes
+        in lieu of writing the test function.  The test function can be overridden
+        by extending classes, if desired.  This test will minimally ensure
+        that sandbox tables are empty and all data is loaded prior to test
+        execution.  It will then ensure only expected cleaned data exists in
+        the cleaned tables, and defined sandboxed data exists in the sandbox
+        table(s).  This class is optional and is not required.  It is here to
+        make testing easier.  All assertions are based on {tablename}_id fields.
+        """
+
+        @classmethod
+        def initialize_class_vars(cls):
+            super().initialize_class_vars()
+            # a list of dictionaries where each dictionary defines table
+            # expectations for each table to validate with rule execution, such as:
```

Comment:
nit: Is this documented elsewhere? If not - it would make sense to have this documentation be exhaustive, rather than "such as" - which indicates this is just a sampling of the valid keys

Reply:
updated documentation.